docs/proposal/20231109-powervs-infra-creation.md

```diff
@@ -0,0 +1,118 @@
+	// serviceInstance can be created via IBM Cloud catalog or CLI.
+	// supported serviceInstance identifier in PowerVSResource are Name and ID and that can be obtained from IBM Cloud UI or IBM Cloud cli.
+	// More detail about Power VS service instance.
+	// https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-creating-power-virtual-server
+	// when omitted system will dynamically create the service instance
+	// +optional
+	ServiceInstance IBMPowerVSResourceReference `json:"serviceInstance"`
+
+	// zone is the name of Power VS zone where the cluster will be created
+	Zone string `json:"zone"`
+
+	// resourceGroup name under which the resources will be created.
+	ResourceGroup string `json:"resourceGroup"`
+
+	// vpc contains information about IBM Cloud VPC resources
+	// +optional
+	VPC IBMVPCResourceReference `json:"vpc,omitempty"`
+
+	// transitGateway contains information about IBM Cloud TransitGateway.
+	// +optional
+	TransitGateway IBMTransitGatewayResource `json:"transitGateway,omitempty"`
+
+	// controlPlaneLoadBalancer is optional configuration for customizing control plane behavior.
+	// +optional
+	ControlPlaneLoadBalancer *VPCLoadBalancerSpec `json:"controlPlaneLoadBalancer,omitempty"`
+}
+
+// IBMPowerVSClusterStatus defines the observed state of IBMPowerVSCluster.
+type IBMPowerVSClusterStatus struct {
+	// ready is true when the provider resource is ready.
+	Ready bool `json:"ready"`
+
+	// serviceInstanceID is the reference to the Power VS service on which the server instance(VM) will be created.
+	ServiceInstanceID *string `json:"serviceInstanceID,omitempty"`
+
+	// networkID is the reference to the Power VS network to use for this cluster.
+	NetworkID *string `json:"networkID,omitempty"`
+
+	// dhcpServerID is the reference to the Power VS DHCP server.
+	DHCPServerID *string `json:"dhcpServerID,omitempty"`
+
+	// vpcID is reference to IBM Cloud VPC resources.
+	VPCID *string `json:"vpcID,omitempty"`
+
+	// vpcSubnetID is reference to IBM Cloud VPC subnet.
+	VPCSubnetID *string `json:"vpcSubnetID,omitempty"`
+
+	// transitGatewayID is reference to IBM Cloud TransitGateway.
+	TransitGatewayID *string `json:"transitGatewayID,omitempty"`
+
+	// ControlPlaneLoadBalancer reference to IBM Cloud VPC Loadbalancer.
+	ControlPlaneLoadBalancer *VPCLoadBalancerStatus `json:"controlPlaneLoadBalancer,omitempty"`
+
+	// Conditions defines current service state of the IBMPowerVSCluster.
+	Conditions capiv1beta1.Conditions `json:"conditions,omitempty"`
+}
+
+// IBMTransitGatewayResource holds the TransitGateway information.
+type IBMTransitGatewayResource struct {
+	Name *string `json:"name,omitempty"`
+	ID   *string `json:"id,omitempty"`
+}
+```
+### Following resources will be created
+1. Power VS workspace
+2. Power VS Network [DHCP service]
+3. VPC :
+4. VPC Subnet
+5. Transit gateway
+6. Loadbalancer
+
+### Cluster creation workflow
+
+![img_4.png](../images/powervs-workflow.png)
```

Comment:
Lets use tools like https://excalidraw.com which allows to save the spec file as well so that someone also can easily modify the content later